core/src/test/resources/org/apache/calcite/test/SqlToRelConverterTest.xml

```diff
@@ -664,6 +664,46 @@ LogicalProject(EMPNO=[$0], JOB=[$2])
 ]]>
     </Resource>
   </TestCase>
+  <TestCase name="testCorrelatedScalarSubQueryInSelectList">
+    <Resource name="planNotExpanded">
+      <![CDATA[
+LogicalProject(correlation=[[$cor0]], DEPTNO=[$7], EXPR$1=[$SCALAR_QUERY({
+LogicalAggregate(group=[{}], EXPR$0=[COUNT()])
+  LogicalProject($f0=[0])
+    LogicalFilter(condition=[>($cor0.DEPTNO, 0)])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+})], EXPR$2=[$SCALAR_QUERY({
+LogicalAggregate(group=[{}], EXPR$0=[COUNT()])
+  LogicalProject($f0=[0])
+    LogicalFilter(condition=[AND(>($cor0.DEPTNO, 0), <($cor0.DEPTNO, 10))])
+      LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+})])
+  LogicalTableScan(table=[[CATALOG, SALES, EMP]])
+]]>
+    </Resource>
+    <Resource name="planExpanded">
+      <![CDATA[
+LogicalProject(DEPTNO=[$7], EXPR$1=[$9], EXPR$2=[$10])
+  LogicalCorrelate(correlation=[$cor1], joinType=[left], requiredColumns=[{7}])
+    LogicalCorrelate(correlation=[$cor0], joinType=[left], requiredColumns=[{7}])
```

Comment:
Are the changes in `SubqueryRemoveRule` done so that you get a different correlate variable here?

If yes can you explain why it is a problem to have the same correlated variable. I am asking this cause we have other plans in the repo where multiple correlates have the same correlation id. Check plans for:

* RelOptRulesTest#testDecorrelateTwoExists
* RelOptRulesTest#testDecorrelateTwoExists#testDecorrelateTwoIns